sys/kern/subr_vmem.c

```diff
@@ -1145,12 +1146,12 @@ vmem_xalloc_nextfit(vmem_t *vm, const vmem_size_t size, vmem_size_t align,
 	if (error == ENOMEM && vmem_try_fetch(vm, size, align, flags))
 		goto retry;
 
+	CHERI_ASSERT_VALID(*addrp);
+	*addrp = (vmem_addr_t)cheri_setboundsexact((void *)*addrp, size);
```

Comment:
No need to do this immediately, but if we switch to tag clearing behavior then we probably want an assertion after the assignment as well.

Reply:
Good point. I think we may also want to rename the assertions. Those currently are only enabled for the purecap kernel but we may want to use them for the hybrid kernel as well in some cases, we could have two variants with CHERI_KERN_ASSERT_* and CHERI_ASSERT_* prefixes